src/isStatelessComponent.js

```diff
@@ -57,7 +64,14 @@ function isReturningJSXElement(path) {
           return
         }
 
-        if (isReturningJSXElement(binding.path)) {
+        // Prevents infinite traverse loop.
+        if (binding.path[traversed]) {
+          return
+        }
+
+        binding.path[traversed] = true
```

Comment:
isnt this check sufficient to prevent recursive calls from crashing this transform? not sure if `iteration` counter is really necessary

Reply:
> if iteration counter is really necessary

It's not, it's just here as a defensive logic. We could very well remove it.